packages/next-admin/src/utils/prisma.ts

```diff
@@ -132,6 +133,55 @@ export const getMappedDataList = async (
     console.error(e);
   }
   data = await findRelationInData(data, dmmfSchema?.fields);
+
+  const listFields = options.model?.[resource]?.list?.fields ?? {};
+
+  data.forEach((item, index) => {
+    Object.keys(item).forEach((key) => {
+      let itemValue;
+
+      if (typeof item[key] === "object" && item[key] !== null) {
+        switch (item[key].type) {
+          case "link":
+            itemValue = item[key].value.label;
+            break;
+          case "count":
+            itemValue = item[key].value;
+            break;
+          case "date":
+            itemValue = item[key].value.toString();
+            break;
+          default:
+            itemValue = item[key].id;
+            break;
+        }
+
+        item[key].__nextadmin_formatted = itemValue;
+      } else if (isScalar(item[key]) && item[key] !== null) {
+        item[key] = {
+          type: "scalar",
+          value: item[key],
+          __nextadmin_formatted: item[key].toString(),
+        };
+        itemValue = item[key].value;
+      }
+
+      if (
+        appDir &&
+        key in listFields &&
+        listFields[key as keyof typeof listFields]?.formatter &&
+        !!itemValue
+      ) {
+        item[key].__nextadmin_formatted = listFields[
+          key as keyof typeof listFields
+          // @ts-expect-error
+        ]?.formatter?.(itemValue ?? item[key]);
+      } else {
+        data[index][key] = item[key];
+      }
+    });
+  });
```

Comment:
So the initial behavior was to pass the `options` object as a prop of the `NextAdmin` component, which was causing issues with the fields with the `formatter` function since those were not serializable. So for the app router, the idea was to run the formatting on the server side. To achieve that, we needed to take the behavior of the [`Cell` component](https://github.com/premieroctet/next-admin/blob/dbb5a3e610439d39fe63a40abeabc19de67f3674/packages/next-admin/src/components/Cell.tsx) and apply it on the server side